src/runtime/analytics-service.js

```diff
@@ -84,21 +96,62 @@ export class AnalyticsService {
     return this.iframe_;
   }
 
+  /**
+   * @return {string}
+   * @private
+   */
+  getQueryString_() {
+    return this.doc_.getWin().location.search.slice(1);
+  }
+
+  /**
+   * @return {string}
+   * @private
+   */
+  getReferrer_() {
+    return parseUrl(this.doc_.getWin().document.referrer).origin;
+  }
+
+  /**
+   * @return {!Promise}
+   */
+  setContext_() {
+    const utmParams = parseQueryString(this.getQueryString_());
+    this.context_.setTransactionId(this.xid_.get());
+    this.context_.setReferringOrigin(this.getReferrer_());
+    const name = utmParams['utm_name'];
+    const medium = utmParams['utm_medium'];
+    const source = utmParams['utm_source'];
+    if (name) {
+      this.context_.setUtmName(name);
+    }
+    if (medium) {
+      this.context_.setUtmMedium(medium);
+    }
+    if (source) {
+      this.context_.setUtmSource(source);
+    }
+    return this.xid_.get().then(id => {
+      this.context_.setTransactionId(id);
+    });
+  }
+
   /**
    * @return {!Promise}
    */
   start() {
     this.doc_.getBody().appendChild(this.getElement());
-    // TODO(sohanirao): setup analyticsContext
     return this.activityPorts_.openIframe(this.iframe_, this.src_,
         this.args_).then(port => {
           this.portResolver_(port);
+          return this.setContext_();
         });
   }
 
   /**
    */
   close() {
+    // TODO(sohanirao): Clean up context?
```

Comment:
No need in this `TODO`. I think we can keep it safely dirty.

Reply:
done